pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/AccessController.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.controller.api.resources;
+
+import com.linkedin.pinot.controller.api.access.AccessControl;
+import com.linkedin.pinot.controller.api.access.AccessControlFactory;
+import com.linkedin.pinot.controller.api.access.AllowAllAccessFactory;
+import java.io.IOException;
+import javax.annotation.Priority;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+
+@PreMatching
+@Provider
+@Priority(2)  // After HeaderAdder
+public class AccessController implements ContainerRequestFilter {
+  private static AccessControlFactory _accessControlFactory = new AllowAllAccessFactory();
+
+  public static void setAccessControlFactory(AccessControlFactory accessControlFactory) {
+    _accessControlFactory = accessControlFactory;
+  }
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    AccessControl accessControl = _accessControlFactory.create();
+    if (!accessControl.allowAccess(requestContext)) {
+      throw new WebApplicationException("Access denied", Response.Status.FORBIDDEN);
```

Comment:
Use ControllerAccessException. It also logs correctly. You can add a custom log for 403 there if you want.

Reply:
You mean ControllerApplicationException?

Reply:
Ah yes.